WordPress/gutenberg
# [RNMobile] Add mobile Spacer component

## Description
Ref to gb-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1436
Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1428

**NOTE**: That PR requires `RangeControl` mobile component -> https://github.com/WordPress/gutenberg/pull/17282 to control spacer height using slider.

That PR adds a mobile `Spacer` component.

**UPDATES**:

* Implemented Accessibility  - double press on `Cell` is focusing the `Slider` once screen reader is enabled

* Corrected dark more [check the screenshot](https://user-images.githubusercontent.com/22746080/67369767-e3034100-f579-11e9-911d-8f8ed1b34e2a.png)

* Typing a value works properly [check the gif](https://user-images.githubusercontent.com/22746080/67405004-70fc1d80-f5b4-11e9-8878-b98d1f4d8f21.gif)

* Styling changes according to @iamthomasbishop comment:
  * outline border-width is set to 2px
  * outline border color is set to blue-30
  * dashed border color is dimmed  (30%)
  * added padding-bottom to cell (16px)
  * added `done` button

iOS: Screenshot [one](https://user-images.githubusercontent.com/22746080/67470800-a6554980-f64e-11e9-960f-c5964dcb877c.PNG) and [two](https://user-images.githubusercontent.com/22746080/67470808-a9503a00-f64e-11e9-9544-3569e59d4c64.PNG)
Android: Screenshot [one](https://user-images.githubusercontent.com/22746080/67513376-fce87500-f69a-11e9-8b10-92f8f860670c.png) and [two](https://user-images.githubusercontent.com/22746080/67513379-feb23880-f69a-11e9-87e0-f73726868f6b.png).

* Fix focusing slider when VO is ON on iPhoneX

* Add an information for VO about current value when changing value



## How has this been tested?

* Choose `Spacer` component.
* Press settings button
* Play with a spacer height using slider:
  * default value: 100
  * minValue: 20
  * maxValue: 500 **NOTE:** if user sets height 800 on the website and then edits it on the phone, the value from web can be in the middle of the range, so user can decrease and increase the value as well. 
* Save post on mobile
* Open the same post on the web
* Play with the height, e.g. set it to 1000 and save a post
* Open the same post again on mobile and check the `Spacer` height

## Screenshots 

android | ios
--- | ---
![spacer_android](https://user-images.githubusercontent.com/22746080/66636940-6b94ef80-ec12-11e9-89af-34cd1b505925.gif) | ![spacer_ios](https://user-images.githubusercontent.com/22746080/66636950-6f287680-ec12-11e9-880d-7c2414999047.gif)

* setting a spacer height on web

![spacer_4](https://user-images.githubusercontent.com/22746080/67196269-c5f43400-f3fa-11e9-8db7-cff10caaa12f.gif)

* setting a spacer height on mobile

![spacer_5](https://user-images.githubusercontent.com/22746080/67196545-53378880-f3fb-11e9-92fa-357150149f1c.gif)


**NOTE:** On the gifs there is a wrong label `Slider` instead of `Height in pixels` which is correct  in the code.


## Types of changes





## Checklist:
